lib/src/Uncertainty/Algorithm/Simulation/NAIS.cxx

```diff
@@ -0,0 +1,240 @@
+  }
+  else
+  {
+    // Computation of weights_
+    weights_ = computeWeights(sample_, outputSample, currentQuantile, initialDistribution_);
+
+    // Computation of auxiliary distribution
+    auxiliaryDistribution = computeAuxiliaryDistribution(sample_, weights_);
+  }
+
+  while ((getEvent().getOperator()(getEvent().getThreshold(), currentQuantile)) && (currentQuantile != getEvent().getThreshold()))
+  {
+    // Drawing of samples using auxiliary density and evaluation on limit state function   
+    sample_ = Sample(0, initialDistribution_.getDimension());
+    outputSample = Sample(0, 1);
+
+    for (UnsignedInteger i = 0; i < getMaximumOuterSampling(); ++ i)
+    {
+      const Sample blockSample(auxiliaryDistribution.getSample(getBlockSize()));
+      sample_.add(blockSample);
+      outputSample.add(getEvent().getFunction()(blockSample));
+      
+      if (stopCallback_.first && stopCallback_.first(stopCallback_.second))
+        throw InternalException(HERE) << "User stopped simulation";
+    } 
+    
+    // Computation of current quantile
+    currentQuantile = outputSample.computeQuantile(rhoQuantile_)[0];
+
+    // If failure probability reached, stop the adaptation
+    if (getEvent().getOperator()(currentQuantile, getEvent().getThreshold()))
+    {
+      currentQuantile = getEvent().getThreshold();
+    }
+    else
+    {
+      // Computation of weights_
+      weights_ = computeWeights(sample_, outputSample, currentQuantile, auxiliaryDistribution);
+
+      // Update of auxiliary distribution
+      auxiliaryDistribution = computeAuxiliaryDistribution(sample_, weights_);
+    }
+    
+    if (stopCallback_.first && stopCallback_.first(stopCallback_.second))
+      throw InternalException(HERE) << "User stopped simulation";
+      
+  } // while
+
+  // Find failure sample indices
+  Indices indicesCritic(0);
+  for (UnsignedInteger i = 0; i < outputSample.getSize(); ++i)
+  {
+    // Find failure Points
+    if (getEvent().getOperator()(outputSample(i, 0), getEvent().getThreshold()))
+      indicesCritic.add(i);
+  } // for i
+
+  const Sample resp_sampleCritic(outputSample.select(indicesCritic));
+  const Sample sampleCritic(sample_.select(indicesCritic));
+
+  // Evaluate initial log PDF in parallel on failure sample
+  Sample logPDFInitCritic(initialDistribution_.computeLogPDF(sampleCritic));
+
+  // Evaluate auxiliary log PDF in parallel on failure sample
+  Sample logPDFAuxiliaryCritic(auxiliaryDistribution.computeLogPDF(sampleCritic));
+
+  Scalar sumPdfCritic = 0.0;
+  for(UnsignedInteger i = 0; i < indicesCritic.getSize(); ++i)
+  {
+    sumPdfCritic += std::exp(logPDFInitCritic(i, 0) - logPDFAuxiliaryCritic(i, 0));
+  }
+
+  // Save of data in Simulation naisResult_ structure
+  naisResult_.setProbabilityEstimate(sumPdfCritic / numberOfSample);
+  naisResult_.setAuxiliarySample(sample_);
+  naisResult_.setAuxiliaryDistribution(auxiliaryDistribution);
+  outputSample_ = outputSample;
```

Comment:
it should also have the number of blocs actually evaluated (outerSampling) and block size